.github/workflows/sql-test-and-build-workflow.yml

```diff
@@ -19,12 +19,20 @@ on:
 
 jobs:
   build:
+    env:
+      BUILD_ARGS: ${{ matrix.os_build_args }}
     strategy:
+      # Run all jobs
+      fail-fast: false
       matrix:
-        java:
-          - 11
-          - 17
-    runs-on: ubuntu-latest
+        java: [11, 17]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        include:
```

Comment:
There's a cleaner way to do this, e.g. https://github.com/opensearch-project/opensearch-java/blob/main/.github/workflows/test-integration.yml#L11

Reply:
I am +1 to Daniel's comment.  I'd rather see the json-looking iterations.   Sometime in the future we may want to change which mixes of jvm+os are supported, and it's much easier to pick and choose as time goes on.

For instance, jvm 20 comes out, but it only builds correctly on {linux} for nearly a year.
Do we _not_ test jvm20 at all until it can build on all three os?  Or do we supply "proof" of tested environemnt for those environments that we can show?    
I favor being able to test what we can when we can.

Thoughts @dblock ?